packages/grid/_modules_/grid/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -282,6 +282,7 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
 
     for (let i = 0; i < renderedRows.length; i += 1) {
       const { id, model } = renderedRows[i];
+      const isLastRow = currentPage.range.firstRowIndex + i === currentPage.rows.length - 1;
```

Comment:
I don't think this formula is giving the correct value. In https://deploy-preview-3519--material-ui-x.netlify.app/components/data-grid/rows/#row-height there're 100 rows and the last row has `data-rowindex=102` when it should end at 99.

<img width="400" src="https://user-images.githubusercontent.com/42154031/152626574-f5c31b31-efa2-41f1-8506-91cf4c445c33.png" alt="image" style="max-width: 100%;">

In #3838 I have a situation where I also need to know if one row is the last one. I solved by using `currentPage.range.firstRowIndex + firstRowToRender + i`: https://github.com/mui-org/material-ui-x/pull/3848/files#diff-7fb7dbb1d7fe2c699d9fc5cfc5c6a56b2489af3b46dc1a2a8eaa3404575cacbdR313

The rationale is:

- `firstRowToRender` gives the first row to render in the current render context
- `i` gives which row we're rendering inside the current render context
- `currentPage.range.firstRowIndex` gives the first row to base with when pagination is on, because in the 2nd page `firstRowToRender` is 0.

Reply:
Yes, makes sense. I'm using `firstRowToRender + i` to pass ´isLastVisible` in https://github.com/mui/mui-x/pull/3848/files#diff-d686bc46bbd2d70d8fe115f2d0f87743bf452fddf658b5dbe8c79d3bd0bb52cbR151